toolkit/tools/specreader/specreader.go

```diff
@@ -315,82 +315,82 @@ func readSpecWorker(requests <-chan string, results chan<- *parseResult, cancel
 			timestamp.StopEvent(ts)
 			ts = nil
 		}
-		ts, _ = timestamp.StartEvent(filepath.Base(specfile), tsRoot)
-
-		result := &parseResult{}
+		ts, _ = timestamp.StartEvent(filepath.Base(specFile), tsRoot)
 
 		providerList := []*pkgjson.Package{}
-		buildRequiresList := []*pkgjson.PackageVer{}
-		sourcedir := filepath.Dir(specfile)
+		sourceDir := filepath.Dir(specFile)
+		testBuildRequiresList := []*pkgjson.PackageVer{}
 
 		// Find the SRPM associated with the SPEC.
-		srpmResults, err := rpm.QuerySPEC(specfile, sourcedir, querySrpm, arch, defines, rpm.QueryHeaderArgument)
+		srpmResults, err := rpm.QuerySPEC(specFile, sourceDir, querySrpm, arch, noCheckDefines, rpm.QueryHeaderArgument)
 		if err != nil {
-			result.err = err
-			results <- result
+			sendEmptyResult(results, err)
 			continue
 		}
 
 		srpmPath := filepath.Join(srpmsDir, srpmResults[0])
 
-		isCompatible, err := rpm.SpecArchIsCompatible(specfile, sourcedir, arch, defines)
+		isCompatible, err := rpm.SpecArchIsCompatible(specFile, sourceDir, arch, noCheckDefines)
```

Comment:
Do you think there is ever a chance a check section is arch dependent?

Reply:
Good point, but I think this should be treated as a spec misconfiguration. If a package builds for a specific architecture, then it should also be testable for it. Right now, the only exception that comes to my mind, is that the test dependencies might not be available for a specific architecture. In such cases I'd lean toward skipping the `%check` section instead of meddling with the `BuildArch` or `ExclusiveArch` tags, since they are related to building the package itself and the author of the spec shouldn't use them to manipulate tests.